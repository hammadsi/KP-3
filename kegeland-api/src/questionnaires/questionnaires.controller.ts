import { Body, Controller, Get, Post, UseGuards } from '@nestjs/common';
import { Delete, Param, Put, Query } from '@nestjs/common/decorators';
import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';

import { FirebaseAuthGuard } from '../firebase/firebase-auth.guard';
import { Role } from '../roles/enums/role.enum';
import { Roles } from '../roles/roles.decorator';
import { RolesGuard } from '../roles/roles.guard';

import { AssignQuestionnaireDto } from './dto/assign-questionnaire.dto';
import { CreateAnswerDto } from './dto/create-answer.dto';
import { CreateQuestionnaireDto } from './dto/create-questionnaire.dto';
import { FindAnswersDto } from './dto/find-answers.dto';
import { FindQuestionnairesDto } from './dto/find-questionnaires.dto';
import { GetAssignedQuestionnaireDto } from './dto/get-assigned-questionnaire.dto';
import { UpdateAnswerDto } from './dto/update-answer.dto';
import { UpdateQuestionnaireDto } from './dto/update-questionnaire.dto';
import { QuestionnairesService } from './questionnaires.service';

@ApiTags('Questionnaires')
@Controller('questionnaires')
@ApiBearerAuth('access-token')
@UseGuards(FirebaseAuthGuard, RolesGuard)
@Roles(Role.PATIENT, Role.PHYSICIAN)
export class QuestionnairesController {
  constructor(private readonly questionnairesService: QuestionnairesService) {}

  /**
   * Endpoint for assigning questionnaire to user
   * @param data fields and data for questionnaire assignment as body of request
   * @returns Promise<AssignQuestionnaireDto>
   */
  @Post('assignments')
  assignQuestionnaire(@Body() data: AssignQuestionnaireDto) {
    return this.questionnairesService.assignQuestionnaire(data);
  }

  /**
   * Endpoint for fetching all assignments per user
   * @param userId
   * @param filters query parameters
   * @returns Promise<{id: string}>
   */
  @Get('assignments/:userId')
  getAssignedQuestionnaire(
    @Param('userId') userId: string,
    @Query() filters: GetAssignedQuestionnaireDto,
  ) {
    return this.questionnairesService.getAssignedQuestionnaire(userId, filters);
  }

  /**
   * Endpoint for fetching all questionnaires
   * @param filters query parameters passed in the URL of the request
   * @returns list of all questionnaires
   */
  @Get()
  findAllQuestionnaires(@Query() filters: FindQuestionnairesDto) {
    return this.questionnairesService.findAllQuestionnaires(filters);
  }

  /**
   * Endpoint for fetching single questionnaire
   * @param id
   * @returns specific questionnaire
   */
  @Get(':id')
  findOneQuestionnaire(@Param('id') id: string) {
    return this.questionnairesService.findOneQuestionnaire(id);
  }

  /**
   * Endpoint for creating new questionnaire
   * @param data questionnaire data passed as body of the request
   * @returns created questionnaire object
   */
  @Post()
  createQuestionnaire(@Body() data: CreateQuestionnaireDto) {
    return this.questionnairesService.createQuestionnaire(data);
  }

  /**
   *
   * @param id
   * @param data as UpdateQuestionnaireDto
   * @returns questionnaire as {name: string, sensor: Sensor, questions: QuestionField[], id: string}
   */
  @Put(':id')
  updateQuestionnaire(
    @Param('id') id: string,
    @Body() data: UpdateQuestionnaireDto,
  ) {
    return this.questionnairesService.updateQuestionnaire(id, data);
  }

  /**
   * Endpoint for deleting questionnaire
   * @param id of questionnaire
   * @returns {Promise<void>}
   */
  @Delete(':id')
  deleteQuestionnaire(@Param('id') id: string) {
    return this.questionnairesService.deleteQuestionnaire(id);
  }

  /**
   * Endpoint for fetching all answers in a questionnaire
   * @param qid questionnaire id
   * @param filters passed as parameters of request URL
   * @returns all answers for questionnaire according to query parameters
   */
  @Get(':questionaireId/answers')
  findAllAnswers(
    @Param('questionaireId') qid: string,
    @Query() filters: FindAnswersDto,
  ) {
    return this.questionnairesService.findAllAnswers(qid, filters);
  }

  /**
   * Endpoint for fertching specific answer in a questionnaire
   * @param qid questionnaireId
   * @param id answerId
   * @returns Answer for questionnaire
   */
  @Get(':questionaireId/answers/:id')
  findOneAnswer(@Param('questionaireId') qid: string, @Param('id') id: string) {
    return this.questionnairesService.getAnswer(qid, id);
  }

  /**
   * Endpoint for creating new answer in a questionnaire
   * @param qid questionnaireId
   * @param data Answer data passed as body in request
   * @returns Answer that has been sucsessfully created
   */
  @Post(':questionaireId/answers')
  createAnswer(
    @Param('questionaireId') qid: string,
    @Body() data: CreateAnswerDto,
  ) {
    return this.questionnairesService.createAnswer(qid, data);
  }

  /**
   * Endpoint for updating existing answer
   * @param qid id of questionnaire
   * @param id id of answer
   * @param data fields and data to be updated, passed as body of request
   * @returns updated Answer
   */
  @Put(':questionaireId/answers/:id')
  updateAnswer(
    @Param('questionaireId') qid: string,
    @Param('id') id: string,
    @Body() data: UpdateAnswerDto,
  ) {
    return this.questionnairesService.updateAnswer(qid, id, data);
  }

  /**
   * Endpoint for deleting answer in questionnaire
   * @param qid id of questionnaire where the answer belongs
   * @param id id answer that is to be deleted
   * @returns Promise<void>
   */
  @Delete(':questionaireId/answers/:id')
  deleteAnswer(@Param('questionaireId') qid: string, @Param('id') id: string) {
    return this.questionnairesService.deleteAnswer(qid, id);
  }
}
