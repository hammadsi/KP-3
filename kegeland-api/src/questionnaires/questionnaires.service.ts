import { NotFoundException, Injectable } from '@nestjs/common';
import { map } from 'lodash';
import { FirebaseService } from 'src/firebase/firebase.service';

import { AssignQuestionnaireDto } from './dto/assign-questionnaire.dto';
import { CreateAnswerDto } from './dto/create-answer.dto';
import { CreateQuestionnaireDto } from './dto/create-questionnaire.dto';
import { FindAnswersDto } from './dto/find-answers.dto';
import { FindQuestionnairesDto } from './dto/find-questionnaires.dto';
import { GetAssignedQuestionnaireDto } from './dto/get-assigned-questionnaire.dto';
import { UpdateAnswerDto } from './dto/update-answer.dto';
import { UpdateQuestionnaireDto } from './dto/update-questionnaire.dto';

@Injectable()
export class QuestionnairesService {
  constructor(private readonly firebaseService: FirebaseService) {}

  /**
   * Function for assigning questionnaire to a user for a given sensor
   * @param data
   * @returns data upon successful request
   */
  async assignQuestionnaire(data: AssignQuestionnaireDto) {
    const { userId, sensor, questionnaireId } = data;
    await this.firebaseService.firestore
      .collection('userDetails')
      .doc(userId)
      .collection('questionnaires')
      .doc(sensor)
      .set({
        questionnaireId,
      });
    return data;
  }

  /**
   * Function for finding questionnaires assigned to a user
   * @param userId id of user
   * @param filters as GetAssignedQuestionnaireDto
   * @returns questionnaire according to query parameters
   */
  async getAssignedQuestionnaire(
    userId: string,
    filters: GetAssignedQuestionnaireDto,
  ) {
    const { sensor } = filters;
    const ref = await this.firebaseService.firestore
      .collection('userDetails')
      .doc(userId)
      .collection('questionnaires')
      .doc(sensor)
      .get();

    try {
      const snapshot = await this.firebaseService.firestore
        .collection('questions')
        .doc(ref.data().questionnaireId)
        .get();
      return { id: snapshot.id, ...snapshot.data() };
    } catch (err) {
      throw new NotFoundException();
    }
  }

  /**
   * Function for finding all questionnaires for a given sensor
   * @param filters query filters
   * @returns list all questionnaires accoring to query parameters (filters)
   */
  async findAllQuestionnaires(filters: FindQuestionnairesDto) {
    let query: any = this.firebaseService.firestore.collection('questions');
    if ('sensor' in filters) {
      query = query.where('sensor', '==', filters.sensor);
    }
    const snapshots = await query.get();
    return map(snapshots.docs, (doc) => ({ id: doc.id, ...doc.data() }));
  }

  /**
   * Function for finding specific questionnaire by its ID
   * @param id
   * @returns specific questionnaire
   */
  async findOneQuestionnaire(id: string) {
    const snapshot = await this.firebaseService.firestore
      .collection('questions')
      .doc(id)
      .get();
    return { id: snapshot.id, ...snapshot.data() };
  }

  /**
   * Function for creating new questionnaire
   * @param data fields and data for new questionnaire
   * @returns created questionnaire (incuding its id)
   */
  async createQuestionnaire(data: CreateQuestionnaireDto) {
    const doc = await this.firebaseService.firestore
      .collection('questions')
      .add(data);

    return { id: doc.id, ...data };
  }

  /**
   * Function for updating questionnaire
   * @param id of questionnaire to be updated
   * @param data fields and data to be updated
   * @returns Questionnaire
   */
  async updateQuestionnaire(id: string, data: UpdateQuestionnaireDto) {
    await this.firebaseService.firestore
      .collection('questions')
      .doc(id)
      .set(data);
    return { id, ...data };
  }

  /**
   * Function for deleting questionnaire
   * @param id of questionnaire to be deleted
   */
  async deleteQuestionnaire(id: string) {
    await this.firebaseService.firestore
      .collection('questions')
      .doc(id)
      .delete();
  }

  /**
   * Function for finding all answers in a questionnaire
   * @param qid id of questionnaire
   * @param filters query parameters by user or session
   * @returns list of all answers according to query parameters (filter)
   */
  async findAllAnswers(qid: string, filters: FindAnswersDto) {
    let query: any = this.firebaseService.firestore
      .collection('questions')
      .doc(qid)
      .collection('answers');
    if ('userId' in filters) {
      query = query.where('userId', '==', filters.userId);
    }
    if ('sessionId' in filters) {
      query = query.where('sessionId', '==', filters.id);
    }
    
    const snapshots = await query.get();
    return snapshots.docs.map((doc) => ({
      id: doc.id,
      createdAt: doc.createTime.seconds,
      ...doc.data(),
    }));
  }

  /**
   * Function for getting answer
   * @param qid questionnaire id
   * @param id id of answer
   * @returns specific answer
   */
  async getAnswer(qid: string, id: string) {
    const snapshot = await this.firebaseService.firestore
      .collection('questions')
      .doc(qid)
      .collection('answers')
      .doc(id)
      .get();
    return { id: snapshot.id, ...snapshot.data() };
  }

  /**
   * Function for creating new answer
   * @param qid id of questionnaire where the answer belongs
   * @param data contents of answer
   * @returns created answer (including its id)
   */
  async createAnswer(qid: string, data: CreateAnswerDto) {
    const doc = await this.firebaseService.firestore
      .collection('questions')
      .doc(qid)
      .collection('answers')
      .add({
        ...data,
      });

    return { id: doc.id, ...data };
  }

  /**
   * Function for updating exising answer
   * @param qid id of questionnaire where answer belong
   * @param id id of answer
   * @param data fields and data to be changed
   * @returns updated answer
   */
  async updateAnswer(qid: string, id: string, data: UpdateAnswerDto) {
    await this.firebaseService.firestore
      .collection('questions')
      .doc(qid)
      .collection('answers')
      .doc(id)
      .set(data);
    return { id, ...data };
  }

  /**
   * Function for deleting answer
   * @param qid id of questionnaire
   * @param id id of answer to be deleted
   */
  async deleteAnswer(qid: string, id: string) {
    await this.firebaseService.firestore
      .collection('questions')
      .doc(qid)
      .collection('answers')
      .doc(id)
      .delete();
  }
}
