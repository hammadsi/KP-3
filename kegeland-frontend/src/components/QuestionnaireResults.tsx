import {
  Box,
  Divider,
  Flex,
  Grid,
  GridItem,
  Heading,
  Progress,
  Text,
  Tooltip,
  useMediaQuery,
} from '@chakra-ui/react';

import {
  Answer,
  Questionnaire,
} from '../state/ducks/questionnaires/questionnaires.interface';

type QuestionnaireResultsProps = {
  questionnaire: Questionnaire | undefined;
  answers: Answer[];
};

const getMinValOffset = (minVal: string) => {
  return `${Math.max(16 - Math.round(minVal.length / 2), 0) * 4}px`;
};

const getMaxValOffset = (maxVal: string) => {
  return `${Math.max(8 - Math.round(maxVal.length / 2), 0) * 4}px`;
};

const QuestionnaireResults: React.FC<QuestionnaireResultsProps> = ({
  questionnaire,
  answers,
}) => {
  const [isGreaterThanLg] = useMediaQuery('(min-width: 62em)');
  const render = () => {
    if (!questionnaire || answers.length < 2) {
      return (
        <Text textAlign="center" fontSize={18} color="gray.500">
          {!questionnaire
            ? 'No questionnaire assigned'
            : 'No answers for the assigned questionnaire'}
        </Text>
      );
    } else {
      return (
        <Box>
          <Heading as="h3" fontSize={20} fontWeight="normal">
            <span style={{ fontWeight: '500' }}>Questionnaire:</span>{' '}
            {questionnaire.name}
          </Heading>
          <Divider marginY={3} />
          <Grid
            templateColumns={
              isGreaterThanLg ? 'repeat(2, 1fr)' : 'repeat(1, 1fr)'
            }
            gap={5}
            marginBottom={5}>
            {questionnaire.questions.map((item, qidx) => (
              <GridItem key={qidx} paddingX={2}>
                <Text fontSize={18} fontWeight="normal">
                  {item.question}
                </Text>
                <Divider marginY={2} borderColor="primary.400" />
                <Flex justifyContent="space-between" marginBottom={1}>
                  <Text
                    fontWeight="semibold"
                    fontStyle="italic"
                    fontSize={14}
                    marginLeft={getMinValOffset(item.minVal)}>
                    {item.minVal}
                  </Text>
                  <Text
                    fontWeight="semibold"
                    fontStyle="italic"
                    fontSize={14}
                    marginRight={getMaxValOffset(item.maxVal)}>
                    {item.maxVal}
                  </Text>
                </Flex>
                {Array.from(Array(2), () => 0).map((_val, idx) => {
                  const answer = answers[idx].answers[qidx];
                  return (
                    <Flex
                      alignItems="center"
                      paddingRight={8}
                      key={`${qidx}-${idx}`}
                      _first={{ marginBottom: 2 }}>
                      <Text width={20} fontWeight="medium" letterSpacing={0.5}>
                        {idx ? 'After' : 'Before'}
                      </Text>
                      <Text fontWeight="semibold">1</Text>
                      <Tooltip hasArrow label={answer + 1}>
                        <Progress
                          textAlign="left"
                          colorScheme="primary"
                          width="full"
                          marginX={3}
                          max={4}
                          height={5}
                          value={answer}
                        />
                      </Tooltip>
                      <Text fontWeight="semibold">5</Text>
                    </Flex>
                  );
                })}
              </GridItem>
            ))}
          </Grid>
        </Box>
      );
    }
  };

  return render();
};

export default QuestionnaireResults;
