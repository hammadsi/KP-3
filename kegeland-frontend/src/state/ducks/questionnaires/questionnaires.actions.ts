import { createAsyncThunk } from '@reduxjs/toolkit';

import { apiCaller } from '../../../utils/apiCaller';

import {
  Answer,
  FetchAssignedQuestionnaireDTO,
  FetchQuestionnaireAnswersDTO,
  Questionnaire,
} from './questionnaires.interface';

export const fetchAssignedQuestionnaire = createAsyncThunk(
  'questionnaires/fetchAssignedQuestionnaire',
  async (data: FetchAssignedQuestionnaireDTO) => {
    const { userId, ...params } = data;
    return apiCaller<Questionnaire>({
      url: `questionnaires/assignments/${userId}`,
      method: 'GET',
      params,
    });
  },
);

export const fetchQuestionnaireAnswers = createAsyncThunk(
  'questionnaires/fetchQuestionnaireAnswers',
  async (data: FetchQuestionnaireAnswersDTO) => {
    const { questionnaireId, ...params } = data;
    return apiCaller<Answer[]>({
      url: `questionnaires/${questionnaireId}/answers`,
      method: 'GET',
      params,
    });
  },
);
