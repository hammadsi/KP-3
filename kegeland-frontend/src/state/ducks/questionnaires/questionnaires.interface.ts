import { SensorType } from '../sensors/sensors.interface';

export type Questionnaire = {
  id: string;
  name: string;
  sensor: SensorType;
  questions: {
    maxVal: string;
    minVal: string;
    question: string;
  }[];
};

export type Answer = {
  userId: string;
  sessionId: string;
  createdAt: number;
  answers: number[];
};

export interface QuestionnaireState {
  loading: boolean;
  error: string | undefined;
  data: Questionnaire[];
  questionnaire: Questionnaire | undefined;
  answers: Answer[];
}

export type FetchAssignedQuestionnaireDTO = {
  userId: string;
  sensor: SensorType;
};

export type FetchQuestionnaireAnswersDTO = {
  questionnaireId: string;
  sessionId: string;
};
