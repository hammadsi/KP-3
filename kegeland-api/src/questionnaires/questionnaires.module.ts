import { Module } from '@nestjs/common';

import { QuestionnairesController } from './questionnaires.controller';
import { QuestionnairesService } from './questionnaires.service';

@Module({
  imports: [],
  controllers: [QuestionnairesController],
  providers: [QuestionnairesService],
})
export class QuestionnairesModule {}
