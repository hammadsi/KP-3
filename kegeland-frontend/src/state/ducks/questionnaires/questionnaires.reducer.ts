import { createSlice } from '@reduxjs/toolkit';

import {
  isFulfilledAction,
  isPendingAction,
  isRejectedAction,
} from '../../../utils/thunk.utils';

import {
  fetchAssignedQuestionnaire,
  fetchQuestionnaireAnswers,
} from './questionnaires.actions';
import { QuestionnaireState } from './questionnaires.interface';

export const initialState: QuestionnaireState = {
  loading: false,
  error: undefined,
  data: [],
  questionnaire: undefined,
  answers: [],
};

const questionnairesSlice = createSlice({
  name: 'questionnaires',
  initialState,
  reducers: {
    clearQuestionnairesState: (state) => {
      state.loading = false;
      state.error = undefined;
      state.questionnaire = undefined;
      state.answers = [];
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchAssignedQuestionnaire.fulfilled, (state, { payload }) => {
        state.questionnaire = payload;
      })
      .addCase(fetchQuestionnaireAnswers.fulfilled, (state, { payload }) => {
        state.answers = payload;
      })
      .addMatcher(
        (action) => isPendingAction(action, 'questionnaires'),
        (state) => {
          state.loading = true;
          state.error = undefined;
        },
      )
      .addMatcher(
        (action) => isFulfilledAction(action, 'questionnaires'),
        (state) => {
          state.loading = false;
          state.error = undefined;
        },
      )
      .addMatcher(
        (action) => isRejectedAction(action, 'questionnaires'),
        (state, { error }) => {
          state.loading = false;
          state.error = error.message;
        },
      );
  },
});

export const { clearQuestionnairesState } = questionnairesSlice.actions;

export default questionnairesSlice.reducer;
