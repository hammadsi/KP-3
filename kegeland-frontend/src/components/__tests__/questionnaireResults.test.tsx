import renderer from 'react-test-renderer';

import { answers } from '../mocks/answers.mock';
import { questionnaire } from '../mocks/questionnaires.mock';
import QuestionnaireResults from '../QuestionnaireResults';

describe('Test questionnaire results', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      value: () => {
        return {
          matches: false,
          addListener: () => {},
          removeListener: () => {},
        };
      },
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <QuestionnaireResults
          answers={answers}
          questionnaire={questionnaire}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly no answers', () => {
    const tree = renderer
      .create(
        <QuestionnaireResults answers={[]} questionnaire={questionnaire} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly no questionnaire', () => {
    const tree = renderer
      .create(<QuestionnaireResults answers={[]} questionnaire={undefined} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
