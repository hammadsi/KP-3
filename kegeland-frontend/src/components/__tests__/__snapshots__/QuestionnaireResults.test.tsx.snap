// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test questionnaire results renders correctly 1`] = `
<div
  className="css-0"
>
  <h3
    className="chakra-heading css-1pomevn"
  >
    <span
      style={
        Object {
          "fontWeight": "500",
        }
      }
    >
      Questionnaire:
    </span>
     
    test questionnaire
  </h3>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-m6jico"
  />
  <div
    className="css-1dm3oea"
  >
    <div
      className="css-tozizp"
    >
      <p
        className="chakra-text css-1vk2i6v"
      >
        how Happy are you
      </p>
      <hr
        aria-orientation="horizontal"
        className="chakra-divider css-gxls9t"
      />
      <div
        className="css-enqbrg"
      >
        <p
          className="chakra-text css-1h6rx2k"
        >
          not good
        </p>
        <p
          className="chakra-text css-1fadmyj"
        >
          very good
        </p>
      </div>
      <div
        className="css-azfv5b"
      >
        <p
          className="chakra-text css-12jt2tl"
        >
          Before
        </p>
        <p
          className="chakra-text css-10ircq3"
        >
          1
        </p>
        <div
          className="css-dcm4eu"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onPointerDown={[Function]}
          onPointerEnter={[Function]}
        >
          <div
            aria-valuemax={4}
            aria-valuemin={0}
            aria-valuenow={5}
            className="css-1jy0d03"
            role="progressbar"
            style={
              Object {
                "width": "125%",
              }
            }
          />
        </div>
        <p
          className="chakra-text css-10ircq3"
        >
          5
        </p>
      </div>
      <div
        className="css-azfv5b"
      >
        <p
          className="chakra-text css-12jt2tl"
        >
          After
        </p>
        <p
          className="chakra-text css-10ircq3"
        >
          1
        </p>
        <div
          className="css-dcm4eu"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onPointerDown={[Function]}
          onPointerEnter={[Function]}
        >
          <div
            aria-valuemax={4}
            aria-valuemin={0}
            aria-valuenow={1}
            className="css-1jy0d03"
            role="progressbar"
            style={
              Object {
                "width": "25%",
              }
            }
          />
        </div>
        <p
          className="chakra-text css-10ircq3"
        >
          5
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test questionnaire results renders correctly no answers 1`] = `
<p
  className="chakra-text css-12b8tya"
>
  No answers for the assigned questionnaire
</p>
`;

exports[`Test questionnaire results renders correctly no questionnaire 1`] = `
<p
  className="chakra-text css-12b8tya"
>
  No questionnaire assigned
</p>
`;
